components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,238 @@
+        try {
+            activeEventPublisherConfigurations = TenantResourceManagerDataHolder.getInstance()
+                    .getEventPublisherService().getAllActiveEventPublisherConfigurations();
+        } catch (EventPublisherConfigurationException e) {
+            log.error(populateMessageWithData(
+                    TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_FETCHING_SUPER_TENANT_EVENT_PUBLISHER_CONFIGURATION,
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain()), e);
+        }
+        return activeEventPublisherConfigurations;
+    }
+
+    /**
+     * This method returns super tenant event stream configurations.
+     *
+     * @return list of event stream configurations.
+     */
+    private List<EventStreamConfiguration> getSuperTenantEventStreamConfigurations() {
+
+        List<EventStreamConfiguration> eventStreamConfigurationList = null;
+        try {
+            eventStreamConfigurationList = TenantResourceManagerDataHolder.getInstance().getCarbonEventStreamService()
+                    .getAllEventStreamConfigurations();
+        } catch (EventStreamConfigurationException e) {
+            log.error(handleServerException(
+                    TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_FETCHING_SUPER_TENANT_EVENT_STREAM_CONFIGURATION,
+                    e, PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain()));
+        }
+        return eventStreamConfigurationList;
+    }
+
+    private void startTenantFlow(int tenantId, String tenantDomain) {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+    }
+
+    private void startSuperTenantFlow() {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        carbonContext.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+        carbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+    }
+
+    /**
+     * This method creates event publisher configurations tenant wise by using super tenant publisher configurations.
+     *
+     * @param activeEventPublisherConfigurations list of active super tenant publisher configurations.
+     */
+    private void createTenantPublisherConfigurationFromSuperTenantConfig(
+            List<EventPublisherConfiguration> activeEventPublisherConfigurations) {
+
+        for (EventPublisherConfiguration eventPublisherConfiguration : activeEventPublisherConfigurations) {
+            try {
+                if (TenantResourceManagerDataHolder.getInstance().getCarbonEventPublisherService()
+                        .getActiveEventPublisherConfiguration(eventPublisherConfiguration.getEventPublisherName())
+                        == null) {
```

Comment:
Rather can't we check contains

Reply:
we will have to create a eventpublisherconfiguration object to check contains

Reply:
fixed

Reply:
if it's an object comparison it's fine